gprofiler/docker_client.py

```diff
@@ -0,0 +1,80 @@
+import logging
+import re
+from typing import Dict, Optional, List
+
+import docker
+
+DOCKER_SYSTEMD_CGROUPS = [re.compile(r"/system.slice/docker-([a-z0-9]{64})\.scope")]
+
+logger = logging.getLogger(__name__)
+
+
+class DockerClient:
+    def __init__(self):
+        try:
+            self._client = docker.from_env()
+        except Exception:
+            logger.warning('Could not initiate the Docker client, so the profiling data will not include the container'
+                           ' names. If you are running the gProfiler in a container, please mount the Docker sock file'
+                           'by running the Docker run command with the following argument: '
+                           '"-v /var/run/docker.sock:/var/run/docker.sock". Otherwise, please open a new issue here: '
+                           'https://github.com/Granulate/gprofiler/issues/new')
+            self._client = None
+
+        self._pid_to_container_name_cache: Dict[int, str] = {}
+        self._container_id_to_name_cache: Dict[str, str] = {}
+
+    def reset_cache(self):
+        self._pid_to_container_name_cache = {}
+        self._container_id_to_name_cache = {}
```

Comment:
Container IDs are unique enough to be cached forever, I think.

Reply:
I'd reset them from time to time just to make sure we don't have a 9 billion keys map.

Reply:
Can use `functools.lru_cache` on the translation function.

Reply:
Pretty much the same code as I need to save the current list of container names but did it anyway 🤷